server.go

```diff
@@ -111,10 +121,24 @@ func (s *Server) HasMethod(method string) bool {
 	return false
 }
 
+// RegisterBeforeFunc registers the specified function as the function
+// that will be called before every request
+func (s *Server) RegisterBeforeFunc(f func(i *RequestInfo)) error {
```

Comment:
I was thinking to add support to have more than one function, but then went back, will remove soon.